pkg/kubecost/cloudcostitem.go

```diff
@@ -0,0 +1,362 @@
+		return nil, fmt.Errorf("cannot merge nil CloudCostItemSets")
+	}
+
+	if that.IsEmpty() {
+		return ccis.Clone(), nil
+	}
+
+	if !ccis.Window.Equal(that.Window) {
+		return nil, fmt.Errorf("cannot merge CloudCostItemSets with different windows")
+	}
+
+	start, end := *ccis.Window.Start(), *ccis.Window.End()
+	result := NewCloudCostItemSet(start, end)
+
+	for _, cci := range ccis.CloudCostItems {
+		result.Insert(cci)
+	}
+
+	for _, cci := range that.CloudCostItems {
+		result.Insert(cci)
+	}
+
+	return result, nil
+}
+
+// GetCloudCostItemSets
+func GetCloudCostItemSets(start time.Time, end time.Time, window time.Duration, integration string) ([]*CloudCostItemSet, error) {
+	windows, err := GetWindows(start, end, window)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build slice of CloudCostItemSet to cover the range
+	CloudCostItemSets := []*CloudCostItemSet{}
+	for _, w := range windows {
+		ccis := NewCloudCostItemSet(*w.Start(), *w.End())
+		ccis.Integration = integration
+		CloudCostItemSets = append(CloudCostItemSets, ccis)
+	}
+	return CloudCostItemSets, nil
+}
+
+// LoadCloudCostItemSets creates and loads CloudCostItems into provided CloudCostItemSets. This method makes it so
+// that the input windows do not have to match the one day frame of the Athena queries. CloudCostItems being generated from a
+// CUR which may be the identical except for the pricing model used (default, RI or savings plan)
+// are accumulated here so that the resulting CloudCostItem with the 1d window has the correct price for the entire day.
+func LoadCloudCostItemSets(itemStart time.Time, itemEnd time.Time, properties CloudCostItemProperties, isK8s bool, cost, credit float64, CloudCostItemSets []*CloudCostItemSet) {
+	totalMins := itemEnd.Sub(itemStart).Minutes()
+	// Disperse cost of the current item across one or more CloudCostItems in
+	// across each relevant CloudCostItemSet. Stop when the end of the current
+	// block reaches the item's end time or the end of the range.
+	for _, ccis := range CloudCostItemSets {
+		// Determine pct of total cost attributable to this window by
+		// determining the start/end of the overlap with the current
+		// window, which will be negative if there is no overlap. If
+		// there is positive overlap, compare it with the total mins.
+		//
+		// e.g. here are the two possible scenarios as simplidied
+		// 10m windows with dashes representing item's time running:
+		//
+		// 1. item falls entirely within one CloudCostItemSet window
+		//    |     ---- |          |          |
+		//    totalMins = 4.0
+		//    pct := 4.0 / 4.0 = 1.0 for window 1
+		//    pct := 0.0 / 4.0 = 0.0 for window 2
+		//    pct := 0.0 / 4.0 = 0.0 for window 3
+		//
+		// 2. item overlaps multiple CloudCostItemSet windows
+		//    |      ----|----------|--        |
+		//    totalMins = 16.0
+		//    pct :=  4.0 / 16.0 = 0.250 for window 1
+		//    pct := 10.0 / 16.0 = 0.625 for window 2
+		//    pct :=  2.0 / 16.0 = 0.125 for window 3
```

Comment:
Is this the exact same logic as the other one? Is it possible to put in a single function and unit test just that, so we can be sure that there isn't implementation drift if some future dev doesn't realize that its duplicated?